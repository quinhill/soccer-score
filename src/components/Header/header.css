.logo {
  margin: 0 0 20px 0;
  padding: 20px;
  font-family: 'Audiowide', cursive;
  color: #00C5B7;
  font-size: 40px;
}

.nav-container {
  width: 100%;
  display: flex;
  justify-content: center;
}

.nav {
  display: flex;
  flex-wrap: grid;
  grid-template-columns: 30% 30% 30%;
  width: 500px;
  height: 50px;
  border-radius: 20px;
  background-color: white;
}

.link {
  text-decoration: none;
  background-color: white;
  width: 100%;
  padding: 15px;
  color: #222222;
  border: none;
}

.link:hover {
  border: 2px solid #00C5B8;
  box-shadow: 0px 2px 10px 0px rgba(0, 197, 184);
}

.leagues-link {
  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;
  border-right: 1px solid #222222;
}

.live-link {
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
  border-left: 1px solid #222222;
}