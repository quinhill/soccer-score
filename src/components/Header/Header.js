import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { fetchScheduleAM } from '../../thunks/fetchSchedules/fetchScheduleAM';
import { fetchScheduleAS } from '../../thunks/fetchSchedules/fetchScheduleAS';
import { fetchScheduleEU } from '../../thunks/fetchSchedules/fetchScheduleEU';
import { fetchLiveAM } from '../../thunks/fetchLive/fetchLiveAM';
import { fetchLiveEU } from '../../thunks/fetchLive/fetchLiveEU';
import { fetchLiveAS } from '../../thunks/fetchLive/fetchLiveAS';
import { fetchLiveScores } from '../../thunks/fetchLive/fetchLiveScores';
import { getFixtures } from '../../apiKeys';
import './header.css';
import { connect } from 'react-redux';
import * as key from '../../apiKeys';
import PropTypes from 'prop-types';


export class Header extends Component {

  componentDidMount() {
    this.handleLiveFetch()
    this.handleScheduleFetch()
  }
  
  handleScheduleFetch = () => {
    this.props.fetchScheduleAM(key.urlAM());
    this.props.fetchScheduleAS(key.urlAS());
    this.props.fetchScheduleEU(key.urlEU());
  }

  handleLiveFetch = () => {
    // this.props.fetchLiveAM(key.urlAMstring);
    // this.props.fetchLiveEU(key.urlEUstring);
    // this.props.fetchLiveAS(key.urlASstring);
    this.props.fetchLiveScores(key.getFixtures);
  }

  render() {
    return (
      <header>
        <Link to='/' className="logo-link">
          <h1 className="logo">
            soccer-scores
          </h1>
        </Link>
      </header>
    )
  }
}

Header.propTypes = {
  fetchScheduleAM: PropTypes.func,
  fetchScheduleAS: PropTypes.func,
  fetchScheduleEU: PropTypes.func,
  // fetchLiveAM: PropTypes.func,
  // fetchLiveAS: PropTypes.func,
  // fetchLiveEU: PropTypes.func,
}

export const mapDispatchToProps = (dispatch) => ({
  fetchScheduleAM: (url) => dispatch(fetchScheduleAM(url)),
  fetchScheduleAS: (url) => dispatch(fetchScheduleAS(url)),
  fetchScheduleEU: (url) => dispatch(fetchScheduleEU(url)),
  // fetchLiveAM: (url) => dispatch(fetchLiveAM(url)),
  // fetchLiveEU: (url) => dispatch(fetchLiveEU(url)),
  // fetchLiveAS: (url) => dispatch(fetchLiveAS(url)),
  fetchLiveScores: (url) => dispatch(fetchLiveScores(url))
});

export default connect(null, mapDispatchToProps)(Header);