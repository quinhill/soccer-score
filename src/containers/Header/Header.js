import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { fetchLiveScores } from '../../thunks/fetchLiveScores';
import { fetchLeague } from '../../thunks/fetchLeague';
import './header.css';
import { connect } from 'react-redux';
import * as key from '../../apiKeys';
import PropTypes from 'prop-types';
import { setDisplay } from '../../actions/setDisplayAction';


export class Header extends Component {

  componentDidMount() {
    this.handleLiveFetch()
  }

  handleLiveFetch = () => {
    this.props.fetchLiveScores(key.getFixtures);
  }

  setDisplay = () => {
    this.props.setDisplay('home')
  }

  render() {
    return (
      <header>
        <Link onClick={this.setDisplay} to='/' className="logo-link">
          <h1 className="logo">
            soccer-scores
          </h1>
        </Link>
      </header>
    )
  }
}

Header.propTypes = {
  fetchLiveScores: PropTypes.func
}

export const mapStateToProps = (state) => ({
  liveScores: state.liveScores
});

export const mapDispatchToProps = (dispatch) => ({
  fetchLiveScores: (url) => dispatch(fetchLiveScores(url)),
  fetchLeague: (url) => dispatch(fetchLeague(url)),
  setDisplay: (clicked) => dispatch(setDisplay(clicked))
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);