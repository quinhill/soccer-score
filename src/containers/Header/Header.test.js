import React from 'react';
import { Header } from './Header';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps } from './Header';

describe('Header', () => {

  let wrapper;
  let mockFetchLiveScores;
  let mockFetchLeague;
  let mockSetDisplay;
  let mockLiveScores;

  beforeEach(() => {
    mockLiveScores = [{}, {}]
    mockFetchLiveScores = jest.fn();
    mockFetchLeague = jest.fn();
    mockSetDisplay = jest.fn();
    wrapper = shallow(<Header
        fetchLiveScores={mockFetchLiveScores}
        fetchLeague={mockFetchLeague}
        setDisplay={mockSetDisplay}
        liveScores={mockLiveScores}
      />)
  })

  it('should call setDisplay in MDTP when setDisplay is called', () => {
    wrapper.instance().setDisplay()

    expect(mockSetDisplay).toHaveBeenCalled()
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('mapStateToProps', () => {

    it('should return a props object with a liveScores array', () => {
      const mockState = {
        liveScores: [{}, {}],
        game: {},
        team: { id: 'teamId' }
      }
      const expected = {
        liveScores: [{}, {}]
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected);
    })
  })

  describe('mapDispatchToProps', () => {

    it('calls dispatch with an fetchLiveScores action  when fetchLiveScores is called', () => {
      const mockDispatch = jest.fn()
      const url = 'http://somestring.com';

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchLiveScores(url)

      expect(mockDispatch).toHaveBeenCalled()
    })

    it('calls dispatch with a setDisplay action when setDisplay is called', () => {
      const mockDispatch = jest.fn()
      const clicked = 'team';

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setDisplay(clicked)

      expect(mockDispatch).toHaveBeenCalled()
    })
  })
})