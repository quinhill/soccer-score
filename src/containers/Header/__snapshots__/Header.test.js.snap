// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    fetchLeague={[Function]}
    fetchLiveScores={[Function]}
    liveScores={
        Array [
            Object {},
            Object {},
          ]
    }
    setDisplay={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="logo-link"
        onClick={[Function]}
        replace={false}
        to="/"
>
        <h1
                className="logo"
        >
                soccer-scores
        </h1>
</Link>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <h1
          className="logo"
>
          soccer-scores
</h1>,
        "className": "logo-link",
        "onClick": [Function],
        "replace": false,
        "to": "/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "soccer-scores",
          "className": "logo",
        },
        "ref": null,
        "rendered": "soccer-scores",
        "type": "h1",
      },
      "type": [Function],
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="logo-link"
          onClick={[Function]}
          replace={false}
          to="/"
>
          <h1
                    className="logo"
          >
                    soccer-scores
          </h1>
</Link>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h1
            className="logo"
>
            soccer-scores
</h1>,
          "className": "logo-link",
          "onClick": [Function],
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "soccer-scores",
            "className": "logo",
          },
          "ref": null,
          "rendered": "soccer-scores",
          "type": "h1",
        },
        "type": [Function],
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
